src/V3Unknown.cpp

```diff
@@ -118,6 +118,24 @@ class UnknownVisitor final : public VNVisitor {
                 = new AstVar{fl, VVarType::MODULETEMP, m_lvboundNames.get(prep), prep->dtypep()};
             m_modp->addStmtsp(varp);
             AstNode* const abovep = prep->backp();  // Grab above point before we replace 'prep'
+            AstNode* currentStmtNode = abovep;
+            while (!VN_IS(currentStmtNode, NodeStmt)) currentStmtNode = currentStmtNode->backp();
+            VNRelinker linkContext;
+            currentStmtNode = currentStmtNode->unlinkFrBackWithNext(&linkContext);
```

Comment:
Some of the terms of the expression might not be known if you move outside the expression, e.g. if inside a Foreach.  I think you need to make an AstExprStmt, and put the assign inside it.

Reply:
Sorry, could you explain this in more detail?
I don’t quite understand how it should be placed in exprStmt, like new AstExprStmt{fl, ?, ?};.

Reply:
Actually, you can't use an AstExprStmt to form a LValue, which seems to be what's needed here.  I guess we will need the statement as you did.  So, in the tests, please add doing one of these assignments inside a Foreach statement, and cleanup if it fails, thanks..

Reply:
Still a little bit confused... But I added a test, in which, assignment is inside a foreach block. Let me know if I understand it wrongly.

To make things clear, let me explain this patch.
For example, in the new test, struct has three fields (a, b, c):

In original process, the assignment would be something like this:
`
In while loop:
temp0.field_a = i;
if(i > 2) cl.struct[i] = temp0;
temp1.field_b = i + 1;
if(i > 2) cl.struct[i] = temp1;
temp2.field_c = i + 2;
if(i > 2) cl.struct[i] = temp2;
i = i + 1;
`
This is why overwriting happens. And our modification is to add an assignment before to preserve the last status. Like this:
`
In while loop:
temp0 = cl.struct[i];
temp0.field_a = i;
if(i > 2) cl.struct[i] = temp0;
temp1 = cl.struct[i];
temp1.field_b = i + 1;
if(i > 2) cl.struct[i] = temp1;
temp2 = cl.struct[i];
temp2.field_c = i + 2;
if(i > 2) cl.struct[i] = temp2;
i = i + 1;
`